### DISCUSSION TITLE
os system requirements for mlx

### DISCUSSION THREAD

**Discussion by palLiHua** [ml-explore/mlx#1133] [Q&A] [answered]
My os system is 13.0, and I have tried mlx version  from 0.13 to 0.8, it all reported:
libmlx.dylib (built for macos 13.5 which is newer than running os)
I wonder if there any ways that I could check the requirements for each version of mlx, cause my model would run on another machine later and its os system is probably different from my machine.
Also, if there any requirements for cpus? My machine is m1pro, and another one is m2 Ultra.

**Comment by awni** [comment, marked as answer]
I'm not sure how you installed MLX, but usually it will install the right version for your OS. MLX supports MacOS >= 13.5.

The only problem is with OS in the range 13.0 to <13.5 since pip doesn't distinguish them (e.g. `pip install mlx`) so it will install a 13.5 binary on a 13.0 machine.

**Reply by palLiHua** [reply]
Thanks a lot for your answer. So if there any way to install suitable MLX for OS 13.0 ? ( I have tried pip and conda install, neither of them works

**Reply by awni** [reply]
No unfortunately, MLX supports 13.5 and up. You could try building from source but there are likely issues as we do not test below 13.5

**Reply by palLiHua** [reply]
okay... appreciate your patient answer
